lib/store/storage.js

```diff
@@ -194,15 +237,99 @@ var uploadFileIfNotExists = module.exports.uploadFileIfNotExists = function(file
         log.info('Possible bad request or S3 params failure. Check if S3 multi part uploads or transfer acceleration is enabled');
         return callback(err);
 
-      } else {
-        // TODO: Error handling will be further improved once metadata collection process is added.
-        return callback(err);
       }
 
+      // TODO: Error handling will be further improved once metadata collection process is added.
+      return callback(err);
+
     } else if (data) {
       log.info('File already exists. Skipping');
       return callback(null, 'File already exists. Skipping');
 
     }
   });
 };
+
+/**
+ * Based on the parameters provided list all the objects within a Bucket Path
+ * This step is a pre-cursor to comparing the list of objects with latest sync files.
+ * If any of the files are out of sync after refresh they will be deleted
```

Comment:
@sandeepmjay Not sure I follow this comment - was part of the text copied from elsewhere?